core/src/main/java/org/apache/iceberg/BaseMetastoreOperations.java

```diff
@@ -63,6 +64,31 @@ protected CommitStatus checkCommitStatus(
       String newMetadataLocation,
       Map<String, String> properties,
       Supplier<Boolean> commitStatusSupplier) {
+    if (metadataLocationCommitted(
+            tableOrViewName, newMetadataLocation, properties, commitStatusSupplier)
+        .orElse(false)) {
+      return CommitStatus.SUCCESS;
+    }
+    return CommitStatus.UNKNOWN;
```

Comment:
Why did you decide to create a new method?

Reply:
Because `checkCommitStatus` never returns `CommitStatus.FAILURE`, which is required so that we know when to throw a CommitFailedException. And since it's kind of a public API, so I added metadataLocationCommitted which returns false if the new metadata location cannot be found.

Reply:
Hmm interesting... But I think we don't have to worry about that in our case, because the double check happens after HMS explicitly returns an exception, which means it has finished processing right?

Reply:
I don't think we can return UNKNOWN if the metadata location is not found. Because UNKNOWN leads to a CommitStateUnknownException and it won't be retried, which basically means concurrent commit is not possible with NoLock mode.

Reply:
> Hmm interesting... But I think we don't have to worry about that in our case, because the double check happens after HMS explicitly returns an exception, which means it has finished processing right?

There could me multiple HMS instances, and another one might still working on serving the previous request

Reply:
> >Hmm interesting... But I think we don't have to worry about that in our case, because the double check happens after HMS explicitly returns an exception, which means it has finished processing right?
> There could me multiple HMS instances, and another one might still working on serving the previous request

But that will fail... Let me think a bit about this.